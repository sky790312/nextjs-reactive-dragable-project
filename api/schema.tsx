export interface PAGE {
  id: number
  name: string
  url: string
}

export interface BLOCK {
  id: number
  name: string
  x: number
  y: number
  o: number
  c: string
}
