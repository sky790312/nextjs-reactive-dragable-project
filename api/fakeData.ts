/**
 * pages should come from somewhere and get all url config.
 */
export const FAKE_PAGES = [
  {
    id: 1,
    name: 'page1',
    url: '/pages/1',
  },
  {
    id: 2,
    name: 'page2',
    url: '/pages/2',
  },
]

/**
 * items should come from api(maybe by page) and get x, y, o style.
 */
export const FAKE_BLOCKS = [
  {
    id: 1,
    name: 'Element 1-1',
    x: 10,
    y: 10,
    o: 1,
    c: '#008800',
  },
  {
    id: 2,
    name: 'Element 1-2',
    x: 60,
    y: 60,
    o: 0.5,
    c: '#008800',
  },
  {
    id: 3,
    name: 'Element 1-3',
    x: 110,
    y: 110,
    o: 1,
    c: '#008800',
  },
  {
    id: 4,
    name: 'Element 2-1',
    x: 30,
    y: 30,
    o: 1,
    c: '#008800',
  },
]
