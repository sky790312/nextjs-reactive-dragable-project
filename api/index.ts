import { FAKE_PAGES, FAKE_BLOCKS } from '@/api/fakeData'
import { PAGE, BLOCK } from '@/api/schema'

export const fetchPages = async () => {
  const data: PAGE[] = await FAKE_PAGES
  return data
}

export const fetchBlocks = async () => {
  const data: BLOCK[] = await FAKE_BLOCKS
  return data
}
