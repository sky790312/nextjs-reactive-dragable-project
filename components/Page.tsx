import React, { useState } from 'react'
import { useRouter } from 'next/router'
import EditableItem from '@/components/EditableItem'
import { PAGE } from '@/api/schema'
import { Dispatch } from 'redux'
import { blockActions } from '@/store/state.block'
import { useDispatch } from 'react-redux'

interface Props {
  page: PAGE
}

const Page: React.FC<Props> = React.memo(({ page }) => {
  const dispatch: Dispatch = useDispatch()
  const { clearCurrentBlock } = blockActions
  const router = useRouter()
  const currentPageId = Number(router.query.id)
  const isCurrentPage = page.id === currentPageId
  const [pageName, setPageName] = useState(page.name)

  const handleClick = () => {
    if (isCurrentPage) {
      return
    }

    dispatch(clearCurrentBlock())
    router.push(page.url)
  }

  const handleEditConfirm = (newName: string) => {
    setPageName(newName)
  }

  return (
    <EditableItem
      text={pageName}
      shouldHighlight={isCurrentPage}
      handleClick={handleClick}
      handleEditConfirm={handleEditConfirm}
    />
  )
})

export default Page
