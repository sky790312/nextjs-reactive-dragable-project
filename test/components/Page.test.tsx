import React from 'react'
import { render, withProviders } from '@/test/testUtils'
import Page from '@/components/Page'
import { FAKE_PAGES } from '@/api/fakeData'

jest.mock('next/router', () => ({
  useRouter() {
    return {
      route: '',
      pathname: '',
      query: '',
      asPath: '',
    }
  },
}))

describe('Page component', () => {
  it('matches snapshot', () => {
    const { asFragment } = render(
      withProviders(<Page page={FAKE_PAGES[0]} />),
      {},
    )
    expect(asFragment()).toMatchSnapshot()
  })
})
