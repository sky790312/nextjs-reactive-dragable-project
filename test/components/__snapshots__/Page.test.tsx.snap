// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page component matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <div
      class="EditableItem__Text-apsj0b-0 jUHTNR"
    >
      page1
    </div>
  </div>
</DocumentFragment>
`;
